### ISSUE TITLE
graphs are not being displayed

### ISSUE THREAD

**Issue by Maximus48p** [mantisbt-plugins/Statistics#1]
Nice Job!
What i have noticed so far:

Within the jpgraph.php file it is referring to file 'jpgraph_gradient.php', but the file does not exist within the jpgraph folder.
So i have installed the complete jpgraph within the library folder.

Also my plugin and library folder are outside the root folder and it seems that all required_once statements doesn't take care about this.

Because of these two i have added the location of the jpgraph folder, like:
`$jgraph_folder = (config_get( 'library_path' ) . 'jpgraph' . DIRECTORY_SEPARATOR);`

and changed (with search and replace) all necessary recuire_once lines from:
`require_once ('plugins/Statistics/jpgraph/jpgraph`
to
`require_once ( $jgraph_folder . 'jpgraph`

**I remember you were able to set the jpgraph folder in the original plugin.**

After that i encountered some DB query issues.
in, almost all, graph files where there is a query which include 'GROUP BY' to build graph needed to be changed.
for example in project-graph.php;
```
		$query = "SELECT count(*) as the_count, {bug}.status, name, project_id
        FROM {bug},  {project}
		WHERE  $specific_where AND {bug}.project_id = {project}.id
        AND date_submitted >= " . $start . " 
        AND date_submitted <= " . $end . " 
        GROUP BY project_id
		ORDER BY the_count desc limit 10" ;
```

changed it to:
```
		$query = "SELECT count(*) as the_count, {bug}.status, name, project_id
        FROM {bug},  {project}
		WHERE  $specific_where AND {bug}.project_id = {project}.id
        AND date_submitted >= " . $start . " 
        AND date_submitted <= " . $end . " 
        GROUP BY project_id, jde_bug_table.status, name
		ORDER BY the_count desc limit 10" ;
```
With help from ChatGPT i found out that i needed to ad some extra expressions to the GROUP BY statement. 
Errors gone after this!

versions:
PHP 7.4.33
MantisBT 2.25.7
DB mysqli 8.1.0, 8.1.0

**Comment by ghost** [comment]
It appeared that not all files were here for jpgraph. I have added those so now that part is complete. Please use version 1.0.2 and try again. All files for this plugin reside within the plugin itself including the files for jpgraph. So tere is no need to make those changes to find jpgraph files/ As for the Group By issue, I do not have that problem + I do not want this grouping which lead to undesirable results in the graph.

**Comment by Maximus48p** [comment]
ok, thx!

to be complete, underneath the error i received from the DB:

Database query failed. Error received from database was #1055: 
Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mantis.jde_bug_table.status' which is not functionally dependent on columns in GROUP BY clause; 
this is incompatible with sql_mode=only_full_group_by for the query: 
```
SELECT count(*) as the_count, mantis_bug_table.status, name, project_id
FROM mantis_bug_table, mantis_project_table
WHERE project_id IN (31,33,43,40,41,39,34,35,37,44,4,15,24,2,23,5,17,20,22,16) AND mantis_bug_table.project_id = mantis_project_table.id
AND date_submitted >= 1672527600
AND date_submitted <= 1705618799
AND mantis_bug_table.status<80 GROUP BY project_id ORDER BY the_count desc limit 10.
```

**Comment by Maximus48p** [comment]
ok, thx!

to be complete, underneath the error i received from the DB:

Database query failed. Error received from database was #1055: 
Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mantis.bug_table.status' which is not functionally dependent on columns in GROUP BY clause; 
this is incompatible with sql_mode=only_full_group_by for the query: 
```
SELECT count(*) as the_count, mantis_bug_table.status, name, project_id
FROM mantis_bug_table, mantis_project_table
WHERE project_id IN (31,33,43,40,41,39,34,35,37,44,4,15,24,2,23,5,17,20,22,16) AND mantis_bug_table.project_id = mantis_project_table.id
AND date_submitted >= 1672527600
AND date_submitted <= 1705618799
AND mantis_bug_table.status<80 GROUP BY project_id ORDER BY the_count desc limit 10.
```

**Comment by ghost** [comment]
Have alook here, you can disable "only_full_group_by"
https://stackoverflow.com/questions/23921117/disable-only-full-group-by

**Comment by Maximus48p** [comment]
I can confirm that it is needed to change the recuire_once lines if you have you're plugin folder outside the root folder.

**Comment by ghost** [comment]
In that case you may face issues with more plugins. Why would you place the plugin folder outside the root?

**Comment by Maximus48p** [comment]
No, most of the plugins goes well.
I have read something about security... so therefore
And since this should be possible i like to use variables as much as possible.
I will update my fork soon....

**Comment by Maximus48p** [comment]
I have updated my github: https://github.com/Maximus48p/Statistics

**Comment by ghost** [comment]
Hi Maximux48p,

There are some issues with your code (for example the path really needs a “\” at the need to function properly.
I have released 1.03 with some additional features which should work for you also.

Kind regards,
Cas
